llvm/include/llvm/IR/IntrinsicsMips.td

```diff
@@ -252,22 +252,22 @@ def int_mips_repl_ph: ClangBuiltin<"__builtin_mips_repl_ph">,
   Intrinsic<[mips_v2q15_ty], [llvm_i32_ty], [IntrNoMem]>;
 
 def int_mips_pick_qb: ClangBuiltin<"__builtin_mips_pick_qb">,
-  Intrinsic<[llvm_v4i8_ty], [llvm_v4i8_ty, llvm_v4i8_ty], [IntrReadMem]>;
+  Intrinsic<[llvm_v4i8_ty], [llvm_v4i8_ty, llvm_v4i8_ty], [IntrWillReturn, IntrReadMem]>;
 def int_mips_pick_ph: ClangBuiltin<"__builtin_mips_pick_ph">,
```

Comment:
Would be good, if tiring, to add tests for all of these. Do they just not have tests in the first place?

Reply:
They do have tests, though the tests aren't affected by the addition of the attribute. I imagine we could something similar to https://github.com/llvm/llvm-project/pull/98968 where we test for DCE, as that should be affected.

Reply:
Added DCE tests